sml/src/sml_number.c

```diff
@@ -24,8 +24,25 @@
 #define SML_BIG_ENDIAN 1
 #define SML_LITTLE_ENDIAN 0
 
-int sml_number_endian();
-void sml_number_byte_swap(unsigned char *bytes, int bytes_len);
+static void sml_number_byte_swap(unsigned char *bytes, int bytes_len) {
+	int i;
+	unsigned char ob[bytes_len];
+
+	memcpy(&ob, bytes, bytes_len);
+
+	for (i = 0; i < bytes_len; i++)
+		bytes[i] = ob[bytes_len - (i + 1)];
+}
+
+static int sml_number_endian() {
```

Comment:
typically you don't need to know that but e.g. use from endian.h functions like be32toh...